pandas/core/base.py

```diff
@@ -1073,7 +1069,7 @@ def unique(self):
         values = self._values
 
         if not isinstance(values, np.ndarray):
-            result = values.unique()
+            result: ArrayLike = values.unique()
```

Comment:
we've got ExtensionArray in the namespace, can make the isinstance check above for EA

Reply:
I've changed the `isinstance` check, but you still need to declare `result` as `ArrayLike`, because the code below that line can return the result as an `np.ndarray` .  Here's the excerpt (as now written):
```python
        if isinstance(values, ExtensionArray):
            result: ArrayLike = values.unique()
            if self.dtype.kind in ["m", "M"] and isinstance(self, ABCSeries):
                # GH#31182 Series._values returns EA, unpack for backward-compat
                if getattr(self.dtype, "tz", None) is None:
                    result = np.asarray(result)
```